### ISSUE TITLE
yum module in ansible 1.9.2 splits up package installs

### ISSUE THREAD

**Issue by aabdnn** [ansible/ansible-modules-core#1897]
Issue Type: Bug Report

Ansible Version:

$ ansible --version
ansible 1.9.2
configured module search path = None

Ansible Configuration:

ANSIBLE_HOST_KEY_CHECKING=no
ANSIBLE_INVENTORY=/Users/anandb/repos/ripe/ansible/hosts
ANSIBLE_GATHERING=smart
ANSIBLE_REMOTE_TEMP=/tmp
ANSIBLE_SSH_CONTROL_PATH=/Users/anandb/.ssh/mux/%%r@%%h:%%p
ANSIBLE_HASH_BEHAVIOUR=merge
ANSIBLE_TIMEOUT=75
ANSIBLE_RETRY_FILES_ENABLED=no
ANSIBLE_SSH_PIPELINING=yes

Environment:

Managing CentOS 6 from OSX 10.10.4

Summary:

Requesting installation of multiple packages, one of which is epel-release, causes the package to be installed in batches instead of together.

Steps To Reproduce:

On a newly-installed CentOS 6 minimal system, run:
- yum: name={{ item }} state=latest
  with_items:
  - ca-certificates
  - epel-release
  - libselinux-python

Expected Results:

Ansible should call yum with all 3 package names, and allow yum to install them all in a single transaction.

Actual Results:

Ansible ca-certificates first, on its own. This also causes installation/update of a bunch of dependent packages. Then runs yum again to run epel-release on its own. Finally, yum tries to install libselinux-python.

As it happens, on this specific server, the time is wrong, and EPEL URLs are https, so yum can't talk to them, and fails. But I don't expect this behaviour. If yum is really supposed to install ca-certificates, epel-release and libselinux-python in a single transaction, then the incorrect time should not affect this task.

**Comment by bcoca** [comment]
cc @abadger

**Comment by abadger** [comment]
This was supposed to be fixed in 1.9.2... Let me test.

**Comment by abadger** [comment]
Oh, I'm sorry -- install was fixed for 1.9.2.  The logic for latest was more complex and so the fix for that didn't make it into 1.9.2.  The latest is fixed in: https://github.com/ansible/ansible-modules-core/pull/1657  along with some changes to what's returned from check-mode.  Just waiting on some final updates to the output of that before merging.  I'll close this as a duplicate of #1657.

**Comment by aabdnn** [comment]
Good to know that the bug has been found and fixed. But it's very annoying.
Such a huge change should not have been introduced for a minor point
release (which should be reserved for bug fixes and not major incompatible
changes).

I said earlier that the change between 1.9.1 and 1.9.2 had broken our
playbooks and I had to rewrite them. And it's still broken :(

**Comment by aabdnn** [comment]
Will this fix go into 1.9.3?

**Comment by abadger** [comment]
@aabdnn You misunderstand, the bug here was present in installed until 1.9.2.  1.9.2 fixed it for state=installed.

Since the changes to perform the same large fix for latest have not been finished yet, they will likely not be in 1.9.3.

**Comment by abadger** [comment]
Some last minute work this morning has put hte changes needed for state=latest into the stable branch.  So it should be in 1.9.3rc2.

**Comment by aabdnn** [comment]
\o/ Yay... thanks!
